### ISSUE TITLE
Docker container killed after uploading PDF file on Apple Silicon (macOS  Sonoma 14.5)

### ISSUE THREAD

**Issue by Daniel-EST** [kermitt2/grobid#1119]
**Description:**

After updating to macOS Sonoma 14.5, I've noticed that a Docker container from image `lfoppiano/grobid:0.8.0` is being killed shortly after sending a PDF file to Grobid. Before macOS Sonoma 14.5 version it used to work as intended. This is happening on both M3 and M1 processors.

The container starts normally when run with the command:
```sh
docker run -d --ulimit core=0 -p 8070:8070 --name grobid lfoppiano/grobid:0.8.0
```

However, it is terminated immediately afterwards uploading a PDF file with the following error message:

```
2024-05-23 11:38:51 [thread 44 also had an error]
2024-05-23 11:38:51 #
2024-05-23 11:38:51 # A fatal error has been detected by the Java Runtime Environment:
2024-05-23 11:38:51 #
2024-05-23 11:38:51 #  SIGSEGV (0xb) at pc=0x00007ffffee7ac84, pid=1, tid=39
2024-05-23 11:38:51 #
2024-05-23 11:38:51 # JRE version: OpenJDK Runtime Environment (17.0.2+8) (build 17.0.2+8-86)
2024-05-23 11:38:51 # Java VM: OpenJDK 64-Bit Server VM (17.0.2+8-86, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
2024-05-23 11:38:51 # Problematic frame:
2024-05-23 11:38:51 # V  [libjvm.so+0xb37c84]  markWord::displaced_mark_helper() const+0x14
2024-05-23 11:38:51 #
2024-05-23 11:38:51 # No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
2024-05-23 11:38:51 #
2024-05-23 11:38:51 # An error report file with more information is saved as:
2024-05-23 11:38:51 # /opt/grobid/hs_err_pid1.log
2024-05-23 11:38:51 #
2024-05-23 11:38:51 # If you would like to submit a bug report, please visit:
2024-05-23 11:38:51 #   https://bugreport.java.com/bugreport/crash.jsp
2024-05-23 11:38:51 #
2024-05-23 11:38:51 
2024-05-23 11:38:51 [error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007fffff5d7602]
```

**Environment:**

* Operating System: macOS Sonoma 14.5
* Docker version: 26.1.1
* Apple Silicon: Yes

**Comment by lfoppiano** [comment]
I @Daniel-EST thank you for your interest in Grobid. Now that you mention it, I've scheduled the update on Sonoma 14.5 tonight on my M2 💦  

Could you try the image `lfoppiano/grobid:0.8.0-arm` [here](https://hub.docker.com/layers/lfoppiano/grobid/0.8.0-arm/images/sha256-79b85da73bae5c2a483e381c1e1231bc73dc0d6b987f16b867a3eb6e8154d7b8?context=explore)? 

For information, you can follow up the background information about this image on #1014

**Comment by m-blasiak** [comment]
@lfoppiano I got the exact same issue. I tried with `lfoppiano/grobid:0.8.0-arm`, as well as [latest-crf](https://hub.docker.com/layers/lfoppiano/grobid/latest-crf/images/sha256-a8024c08d0e9ff6af678b99a4233eee1f1aaab2719d330236b8bb374189a924a?context=explore) unfortunately the issue persists.

**Comment by Daniel-EST** [comment]
@lfoppiano, just tested the `lfoppiano/grobid:0.8.0-arm` version and the same problem persists. However, while going through the [docker images tags](https://hub.docker.com/r/lfoppiano/grobid/tags) `lfoppiano/grobid:latest-full` worked as intended. Where do I get more information about the differences between both versions?

**Comment by lfoppiano** [comment]
OK, so the `latest-*` images are generally for development. 

In this case you're lucky because it's the same as `lfoppiano/grobid/0.8.0-full-slim` [here](https://hub.docker.com/layers/lfoppiano/grobid/0.8.0-full-slim/images/sha256-b1fbe47707bef24e547fc396437529fd07bbfc38407242c46605d7be55e4be1b?context=explore)

This image was built from the branch `0.8.0-fixes` which is the version 0.8.0 with post-release fixes. 

could you try to run again the `lfoppiano/grobid:0.8.0` or `lfoppiano/grobid:0.8.0-arm` and specify the `--init` param, and let me know if there is any change?

**Comment by Daniel-EST** [comment]
> could you try to run again the `lfoppiano/grobid:0.8.0` or `lfoppiano/grobid:0.8.0-arm` and specify the `--init` param, and let me know if there is any change?

Running
```docker run --init --ulimit core=0 -p 8070:8070 --name grobid lfoppiano/grobid:0.8.0-arm```
The error persisted, with the same trace as described earlier. However, running:

```docker run --init --ulimit core=0 -p 8070:8070 --name grobid lfoppiano/grobid:0.8.0-full-slim```
I was able to upload an PDF file.

**Comment by lfoppiano** [comment]
OK, I'm glad you have a working image, but I don't know why it works... Maybe something has changed in the way java is handled 🤔 

About the `lfoppiano/grobid:0.8.0-full-slim`, does it work well with more than one PDF? Could you run it without issues on let's say 1000 PDF documents?

**Comment by lfoppiano** [comment]
@Daniel-EST did you manage to fix the issue? I've built a crf-only image that should support ARM, but I haven't tested. If you have time, it's on #1165 .

**Comment by lfoppiano** [comment]
I think you should add also `--init` or the child process won't be cleaned up. I tried without and the JVM was crashing. 
After adding `--init` it seems more stable, but I would need to test it a bit more. 

I've added some documentation on the related branch, [here](https://github.com/kermitt2/grobid/blob/multi-arch-docker-image/doc/Grobid-docker.md#arm64).

**Comment by lfoppiano** [comment]
the https://github.com/grobidOrg/grobid/pull/1165 has been merged into master
